tests/unit/handlers/test__monitored_resources.py

```diff
@@ -353,3 +343,40 @@ def test_detect_partial_data(self):
             # project id not returned from metadata serve
             # should be empty string
             self.assertEqual(resource.labels["project_id"], "")
+
+
+class Test_Add_Environmental_Labels(Test_Resource_Detection):
+    def setUp(self):
+        super().setUp()
+        self.record = logging.LogRecord("logname", None, None, None, "test", None, None)
+
+    def add_environment_labels_test_flow(self, resource_type, expected_labels):
+        resource = Resource(type=resource_type, labels={})
+        actual_labels = add_environment_labels(resource, self.record)
+        self.assertDictEqual(actual_labels, expected_labels)
```

Comment:
Maybe this is just me, but I don't like this kind of indirection for assertions. I think this is one place where DRY doesn't apply. It feels to easy to accidently make all tests into no-ops

I prefer to have asserts in each test function (and not behind any conditionals if possible).